scripts/shared/cloud-prepare.sh

```diff
@@ -27,18 +27,55 @@ function check_gateway_exists() {
     [[ $(kubectl get nodes -l "${GATEWAY_LABEL}" --no-headers | wc -l) -gt 0 ]]
 }
 
-function prepare_kind() {
-    read -r -a nodes <<< "${cluster_nodes[$cluster]}"
-    local node=${cluster}-${nodes[-1]}
-    kubectl label node "$node" "$GATEWAY_LABEL" --overwrite
+function check_and_apply_air_gap_env() {
+    local node="$1"
+    local pub_ip
 
     if [[ "$AIR_GAPPED" = true ]]; then
-        local pub_ip
+        echo "Apply Air Gap environment configuration"
         pub_ip=$(kubectl get nodes "$node" -o jsonpath="{.status.addresses[0].address}")
         kubectl annotate node "$node" gateway.submariner.io/public-ip=ipv4:"$pub_ip"
     fi
 }
 
+function prepare_kind() {
+    read -r -a nodes <<< "${label_gw_nodes[$cluster]}"
+
+    worker_count=$(grep -o worker <<< "${nodes[*]}" | wc -l || true)
+    master_count=$(grep -o control-plane <<< "${nodes[*]}" | wc -l || true)
```

Comment:
Do we care if the labeled node is control plane or worker?

If not, it would be simpler to have something like `gateways: 2` and let shipyard choose which 2 nodes to label.

Reply:
@mkolesnik 
I would say we do care.
Normally, the gw node should be always a worker node.
And just in some outstanding cases place it on control-plane.

Reply:
Can you show an example in docs or code where that distinction actually matters?

AFAICT we don't care where the GW node is, the only thing we care about is that it had the correct infra support (ports can be opened, public IP, etc).

Going with a count instead of "worker/control-plane" distinction would be easier to use and maintain.

Reply:
Changed to set the parameter as `gateways: x`.